src/tokenizer/mod.rs

```diff
@@ -227,18 +227,21 @@ impl<Sink: TokenSink> Tokenizer<Sink> {
 
     /// Feed an input string into the tokenizer.
     pub fn feed(&mut self, input: String) {
-        if input.len() == 0 {
+        if input.len() == 0 && self.input_buffers.is_empty() {
             return;
         }
 
-        let pos = if self.discard_bom && input.as_slice().char_at(0) == '\u{feff}' {
-            self.discard_bom = false;
-            3  // length of BOM in UTF-8
-        } else {
-            0
-        };
+        if !input.is_empty() {
+            let pos = if self.discard_bom && input.as_slice().char_at(0) == '\u{feff}' {
+                self.discard_bom = false;
+                3  // length of BOM in UTF-8
+            } else {
+                0
+            };
+
+            self.input_buffers.push_back(input, pos);
+        }
 
-        self.input_buffers.push_back(input, pos);
         self.run();
     }
```

Comment:
Why? It makes the code in Servo easier, since it doesn't have to special-case restarting parsing when no input is available vs. feeding a new block of input that has been received.